WhaleSpotting/react-app/src/components/Profile.tsx

```diff
@@ -57,6 +83,7 @@ export function Profile(): JSX.Element {
             fetchCurrentUserSightings(page)
                 .then(data => setData(data));
         }
+        assignRank();
```

Comment:
Can we call this after we call getUser() instead, as in do getUser().then(() => assignRank()) on line24.

Reply:
We found a problem that the page re-render which cause the Rank to somehow reset to newbie. 
I have temporary commented out and replace it with getRankSrc function.
But it now displays the correct badges.